extendr-api/src/wrapper/matrix.rs

```diff
@@ -131,14 +127,27 @@ impl<T> Offset<[usize; 3]> for RArray<T, [usize; 3]> {
     }
 }
 
-impl<T, D> RArray<T, D> {
-    pub fn from_parts(robj: Robj, data: *mut T, dim: D) -> Self {
-        Self { robj, data, dim }
+impl<'a, T, D> RArray<T, D>
+where
+    T: 'a,
+    Robj: AsTypedSlice<'a, T>,
+{
+    pub fn from_parts(robj: Robj, dim: D) -> Self {
+        Self {
+            robj,
+            dim,
+            _data: std::marker::PhantomData,
+        }
+    }
```

Comment:
Important change to document: 

This is a potentially breaking change for anyone whose code uses the RArray::from_parts()